### PR TITLE
[Do Not Merge] Brownie artifacts for SubscriptionManager deployment on Matic

### PR DESCRIPTION
**Pull request by cygnusv**
- Improved deployment script
- Brownie deployment metadata for SubscriptionManager contracts

We still need to decide on https://github.com/nucypher/nucypher-contracts/issues/18, but for the moment this is the raw deployment information

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,12 @@ __pycache__
 .history
 .hypothesis/
 .tox/
-build/
```
Comment (theref) [inline .gitignore:6]: This is pretty unusual, will each deployment need to be added to the `.gitignore`? I guess this links with #18

File: scripts/deploy_subscription_manager.py [hunk 2]
```diff
@@ -20,15 +20,27 @@ def main(account_name : str = 'test'):
...
     calldata = subscription_manager_logic.initialize.encode_input(INITIAL_FEE_RATE)
-    transparent_proxy = oz.TransparentUpgradeableProxy.deploy(
+    constructor_args = (
         subscription_manager_logic.address,
         proxy_admin.address,
-        calldata,
-        {'from': deployer})
+        calldata
+    )
+    transparent_proxy = deployer.deploy(oz.TransparentUpgradeableProxy,
```
Comment (theref) [inline scripts/deploy_subscription_manager.py:41]: This is quite a long line, could we run `black` on this file

**Review by theref — APPROVED** [review]
A couple of comments, but nothing that would prevent merging

File: build/deployments/137/0x16E74C81A68288B4d216720B65C3E4CF05E2E08d.json [hunk 3]
```diff
@@ -0,0 +1 @@
...be the admin of `proxy`.\n     */\n    function upgrade(TransparentUpgradeableProxy proxy, address implementation) public virtual onlyOwner {\n        proxy.upgradeTo(implementation);\n    }\n\n    /**\n     * @dev Upgrades `proxy` to `implementation` and calls a function on the new implementation. See\n     * {TransparentUpgradeableProxy-upgradeToAndCall}.\n     *\n     * Requirements:\n     *\n     * - This contract must be the admin of `proxy`.\n     */\n    function upgradeAndCall(\n        TransparentUpgradeableProxy proxy,\n        address implementation,\n        bytes memory data\n    ) public payable virtual onlyOwner {\n        proxy.upgradeToAndCall{value: msg.value}(implementation, data);\n    }\n}\n", "sourceMap": "435:2404:139:-:0;;;;;;;;;;;;-1:-1:-1;921:32:4;719:10:196;921:18:4;:32::i;:::-;435:2404:139;;2270:187:4;2343:16;2362:6;;-1:-1:-1;;;;;2378:17:4;;;-1:-1:-1;;;;;;2378:17:4;;;;;;2410:40;;2362:6;;;;;;;2410:40;;2343:16;2410:40;2333:124;2270:187;:::o;435:2404:139:-;;;;;;;", "sourcePath": "contracts/proxy/transparent/ProxyAdmin.sol", "type": "contract", "deployment": {"address": "0x16E74C81A68288B4d216720B65C3E4CF05E2E08d", "chainid": "137", "blockHeight": 25163508}}
```
Comment (manumonti) [inline build/deployments/137/0x16E74C81A68288B4d216720B65C3E4CF05E2E08d.json:1]: What is the purpose of these empty sons?

**Review by manumonti — APPROVED** [review]
Just a question. The contribution is ok in my opinion

File: build/deployments/137/0x16E74C81A68288B4d216720B65C3E4CF05E2E08d.json (hunk 3 above)
Comment (cygnusv) [reply build/deployments/137/0x16E74C81A68288B4d216720B65C3E4CF05E2E08d.json:1]: They're not empty. They have a lot of information: ABI, address, bytecode, etc.

Comment (cygnusv) [reply build/deployments/137/0x16E74C81A68288B4d216720B65C3E4CF05E2E08d.json:1]: The raw data:
https://raw.githubusercontent.com/nucypher/nucypher-contracts/17072d51535f58f801afed8af1c738825b6529da/build/deployments/137/0x16E74C81A68288B4d216720B65C3E4CF05E2E08d.json

Comment (manumonti) [reply build/deployments/137/0x16E74C81A68288B4d216720B65C3E4CF05E2E08d.json:1]: Ok! Got it

**Review by manumonti — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -3,7 +3,12 @@ __pycache__
...
+build/*
+!build/deployments/
+build/deployments/*
```
Comment (derekpierre) [inline .gitignore:8]: This is duplicated no? 

Is it that you want to exclude `build/deployments` and then make specific exceptions for `build/deployments/map.json` , `build/deployments/1`, `build/deployments/137`. If so, then L7 (`!build/deployments/`) can probably go away.

Comment (derekpierre) [reply .gitignore:8]: ^ Ping @cygnusv . More broadly, PR needs a rebase.

**Comment by KPrasch** [comment]
Obsoleted by #119

**Pull request closed without merging**